drivers/include/pcf857x.h

```diff
@@ -0,0 +1,557 @@
+    PCF857X_ERROR_INT_PIN,      /**< interrupt pin initialization failed */
+} pcf857x_error_codes_t;
+
+/**
+ * @brief Definition of PCF857X expander variants
+ *
+ * It is used in configuration parameters to specify the PCF857X expander
+ * used by device.
+ *
+ * @note Expander variants known by the driver depend on enabled pseudomodules
+ * `pcf8574`, `pcf8574a` and `pcf8575`.
+ */
+typedef enum {
+#if IS_USED(MODULE_PCF8574) || DOXYGEN
+    PCF857X_EXP_PCF8574,    /**< PCF8574 8 bit I/O expander used */
+#endif
+#if IS_USED(MODULE_PCF8574A) || DOXYGEN
+    PCF857X_EXP_PCF8574A,   /**< PCF8574A 8 bit I/O expander */
+#endif
+#if IS_USED(MODULE_PCF8575) || DOXYGEN
+    PCF857X_EXP_PCF8575,    /**< PCF8575 16 bit I/O expander */
+#endif
+    PCF857X_EXP_MAX,
+} pcf857x_exp_t;
+
+/**
+ * @brief   PCF857X device initialization parameters
+ */
+typedef struct {
+
+    i2c_t    dev;       /**< I2C device (default I2C_DEV(0)) */
+    uint16_t addr;      /**< I2C slave address offset to the PCF7857X base
+                             address (default 0) */
+    pcf857x_exp_t exp;  /**< PCF857X expander variant used by the device
+                             (default depends on used pseudomodules */
+
+#if IS_USED(MODULE_PCF857X_IRQ) || DOXYGEN
+    gpio_t   int_pin;   /**< MCU interrupt pin or #GPIO_UNDEF if not used (default).
+                             Using interrupt pin has the advantage that inputs
+                             have to be read from expander only if any input
+                             value changes.
+                             @note To use interrupts for expander inputs, this
+                             pin has to be defined. */
+#endif /* MODULE_PCF857X_IRQ */
+} pcf857x_params_t;
+
+#if IS_USED(MODULE_PCF857X_IRQ) || DOXYGEN
+/**
+ * @brief   IRQ event type
+ *
+ * Handling an interrupt of a PCF857x expander requires the driver to access
+ * the device directly via I2C. However, the mutex-based synchronization of
+ * I2C accesses does not work in the interrupt context. Therefore the ISR must
+ * not access the PCF857x expander device directly. Rather, the ISR must only
+ * indicate the occurrence of the interrupt which has to be handled
+ * asynchronously in the thread context.
+ *
+ * The type defines the data structure that is part of each device data
+ * structure to indicate that an interrupt of the device occurred. Since there
+ * is only one interrupt source, only one interrupt can be pending per device.
+ * Thus, only one object of this type per device is required.
+ */
+typedef struct {
+    event_t event;      /**< inherited event data structure */
+    void *dev;          /**< PCF857X device reference */
+} pcf857x_irq_event_t;
+
+#endif /* MODULE_PCF857X_IRQ */
+
+/**
+ * @brief   PCF857X device data structure type
+ */
+typedef struct {
+    pcf857x_params_t params;  /**< device initialization parameters */
+
+    unsigned int pin_num;     /**< number of I/O pins, depends on used expander
```

Comment:
```suggestion
    uint8_t pin_num;     /**< number of I/O pins, depends on used expander
```
should suffice

Reply:
maybe `uint_fast8_t`?

Reply:
Sure, although it seems like it's mostly used in asserts? I guess you can check the code size to see what makes most sense.

Reply:
Used `uint8_t` because it saves some bytes on STM32 without increasing the code.